docs/coordinates/inplace.rst

```diff
@@ -0,0 +1,35 @@
+In Place Modification of Coordinates
+====================================
+
+Coordinates are generally considered to be immutable. If you want to create
+another coordinate frame with different data you should use
+`~astropy.coordinates.BaseCoordinateFrame.realize_frame`, this is the safest way
+to change the data in a frame as it creates a new frame with that data.
+
+Due to the fact that this creates a new frame it is quite slow, and in some
+situations it is required that the data is replaced inside a frame with minimal
+changes to the frame for maximum speed. This modification can be done by
+modifiying the values of the representation data as follows::
+
+    c = SkyCoord([1,2],[3,4], unit='deg')
+    c.data.lon[:] = [10, 20] * u.deg
+
+This changes the longitude values for the frame, however
+`~astropy.coordinates.SkyCoord` and `~astropy.coordinates.BaseCoordinateFrame`
+cache representations of the data held within the frame as they are calculated
+to speed up future operations of different representations of the same data. If
+you modify the data in the representation the frame contains it is important to
+clear this cache so future calculations of new representations are recomputed
+using the new data. This can be achieved by doing::
+
+      del c.frame.cache['representation']
```

Comment:
If you write `del c.cache['representation']` then the example should work for either a SkyCoord or Frame.  (I *think* so anway, needs to be verified).